src/Sarif.Viewer.VisualStudio.Interop/SarifViewerInterop.cs

```diff
@@ -0,0 +1,190 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Linq;
+using System.Reflection;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.Sarif.Viewer.Interop
+{
+    public class SarifViewerInterop
+    {
+        private static readonly string ViewerAssemblyFileName = "Microsoft.Sarif.Viewer";
+        private static readonly string ViewerServiceInterfaceName = "SLoadSarifLogService";
+        private bool? _isViewerExtensionInstalled;
+        private bool? _isViewerExtensionLoaded;
+        private Assembly _viewerExtensionAssembly;
+        private AssemblyName _viewerExtensionAssemblyName;
+        private Version _viewerExtensionVersion;
+
+        /// <summary>
+        /// Gets the Visual Studio shell instance object.
+        /// </summary>
+        public IVsShell VsShell { get; private set; }
+
+        /// <summary>
+        /// Gets the unique identifier string of the SARIF Viewer package.
+        /// </summary>
+        public static readonly string ViewerExtensionGuidString = "b97edb99-282e-444c-8f53-7de237f2ec5e";
+
+        /// <summary>
+        /// Gets the unique identifier of the SARIF Viewer package.
+        /// </summary>
+        public static readonly Guid ViewerExtensionGuid = new Guid(ViewerExtensionGuidString);
+
+        #region Private properties for lazy initialization
+        private Assembly ViewerExtensionAssembly
+        {
+            get
+            {
+                if (_viewerExtensionAssembly == null)
+                {
+                    Assembly[] assemblies = AppDomain.CurrentDomain.GetAssemblies();
+                    _viewerExtensionAssembly = assemblies.Where(a => a.GetName().Name == ViewerAssemblyFileName).FirstOrDefault();
+                }
+
+                return _viewerExtensionAssembly;
+            }
+        }
+
+        private AssemblyName ViewerExtensionAssemblyName
+        {
+            get
+            {
+                return _viewerExtensionAssemblyName ?? (_viewerExtensionAssemblyName = ViewerExtensionAssembly.GetName());
+            }
+        }
+
+        private Version ViewerExtensionVersion
+        {
+            get
+            {
+                return _viewerExtensionVersion ?? (_viewerExtensionVersion = ViewerExtensionAssemblyName.Version);
+            }
+        }
+        #endregion
+
+        private SarifViewerInterop() { }
```

Comment:
>SarifViewerInterop [](start = 16, length = 18)

Since there is a non-default ctor, the compiler won't supply a default ctor, so there's no danger of anyone accidentally default-constructing a `SarifViewerInterop`. So I don't think you need?

Reply:
Right, forgot about that

---
In reply to: [211044821](https://github.com/Microsoft/sarif-visualstudio-extension/pull/9#discussion_r211044821) [](ancestors = 211044821)